bin/oneoff/backfill_data/backfill_script_announcement_keys.rb

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env ruby
+
+# In order to unqiuely identify the announcements associated
+# with a Script, we are generating a UUID key. This can then
+# be used to accurately look up the translation amongst a
+# Script's array of announcements.
+
+require_relative '../../../dashboard/config/environment'
+
+def backfill_script_announcement_keys
+  Script.all.each do |script|
+    next unless script.announcements
+
+    # Create UUID key for each announcement
+    script.announcements.each do |announcement|
+      announcement[:key] = SecureRandom.uuid unless announcement[:key]
+    end
+    begin
+      script.save
```

Comment:
you will need `save!` here in order to get an exception

Reply:
Ah I removed `save!` testing something out good catch. Will fix it in a moment.